C/common/reading_set.cpp

```diff
@@ -192,6 +192,51 @@ ReadingSet::append(const vector<Reading *>& readings)
 	}
 }
 
+/**
+* Deep copy a set of readings to this reading set.
+*/
+bool
+ReadingSet::copy(const ReadingSet& src)
+{
+   vector<Reading *> *readings = new vector<Reading *>;
+   bool copyResult = true;
+   try
+   {
+	   for ( auto rs : src.getAllReadings())
+	   {
+		   std::string assetName = rs->getAssetName();
+		   for ( auto dp : rs->getReadingData())
+		   {
+			   std::string dataPointName  = dp->getName();
+			   DatapointValue dv = dp->getData();
+			   Datapoint *value = new Datapoint(dataPointName, dv);
+			   Reading *in = new Reading(assetName, value);
+			   readings->push_back(in);
+		   }
+	   }
+   }
+   catch (std::bad_alloc& ex)
+   {
+		copyResult = false;
+		readings->clear();
+		Logger::getLogger()->error("Insufficient memory :: failed while copying :%d: reading from ReadingSet :%s: ",readings->size()+1, ex.what());
```

Comment:
You don't need the +1, size() returns the number of items in the container. Also please loose the extra :'s you are using to surround things. Error level messages are meant for users, so should be well formatted statements.

Reply:
Yes, size() returns the number but +1 will give current items which was being copied when exception occurred and not  collected so far into readings vector.  does it make correct use case to include +1 with size() ?